kod/object/item/passitem/necklace/shadowam.kod

```diff
@@ -236,28 +230,21 @@ messages:
    RevealHiddenAttributes()
    {
       local bDone;
-
       bDone = FALSE;
 
-      % Reveal the true name and description of the item.
-      if vrName <> vrRealName
+      if pbIdentified = FALSE
       {
-         vrName = vrRealName;
-         vrDesc = vrRealDesc;
-         bDone = TRUE;
+         pbIdentified = TRUE;
+         Send(poOwner, @SomethingChanged, #what=self);
       }
-      
+
       return bDone;
    }
 
-   GetTrueName()
-   {
-      return vrRealName;
-   }
 
    IsIdentified()
    {
```

Comment:
This function can be written simply
```
    return pbIdentified;
```

Reply:
Done.  Embarrassing.  I was going to fix this but you beat me to it.  Much better.